sidecar/integration/azure_intergration_test.go

```diff
@@ -0,0 +1,272 @@
+		t.Errorf("error reading out dir '%+v'", err)
+	}
+	outLength := len(outFiles)
+
+	// Write an output metadata file
+	insight := []byte(`[{"key": "key2","value": "value2"}]`)
+	writeOutputBytes(insight, outMetaFilePath)
+
+	// Write an output event file
+	j := fmt.Sprintf(`[{"key":"eventType","value":"%s"},{"key":"files","value":"%s,%s"},{"key":"abc","value":"123"}]`, eventTypes[0], blob1, blob2)
+	outEvent := []byte(j)
+
+	writeOutputBytes(outEvent, outEventFilePath)
+
+	client := &http.Client{}
+
+	// Ready will attempt to sync the execution environment for this module - this should be empty
+	if err := executeRequest(client, config.SharedSecret, fmt.Sprintf("%v", config.ServerPort), "ready"); err != nil {
+		t.Errorf("error calling ready '%+v'", err)
+	}
+
+	// Done will commit the written files to external providers
+	if err := executeRequest(client, config.SharedSecret, fmt.Sprintf("%v", config.ServerPort), "done"); err != nil {
+		t.Errorf("error calling done '%+v'", err)
+	}
+
+	// Clear local module directories
+	module1.Close()
+
+	// Hydrate event
+	b, err := ioutil.ReadFile(inEventFilePath)
+	if err != nil {
+		t.Errorf("error reading event from disk '%+v'", err)
+	}
+	var inEvent common.Event
+	err = json.Unmarshal(b, &inEvent)
+	if err != nil {
+		t.Errorf("error unmarshalling event '%+v'", err)
+	}
+
+	// Create Module #2
+	config.Context.ParentEventID = config.Context.EventID
+	config.Context.EventID = inEvent.EventID
+	module2, err := createModule(config)
+	if err != nil {
+		t.Error(err)
+	}
+	defer module2.Close()
+
+	// Ready will attempt to sync the execution environment for this module.
+	// This should download the files written by the previous done.
+	if err := executeRequest(client, config.SharedSecret, fmt.Sprintf("%v", config.ServerPort), "ready"); err != nil {
+		t.Errorf("error calling done '%+v'", err)
+	}
+
+	// Check blob input data matches the output from the first module
+	inFiles, err := ioutil.ReadDir(inDataDir)
+	if err != nil {
+		t.Errorf("error reading in dir '%+v'", err)
+	}
+	inLength := len(inFiles)
+
+	if (inLength != outLength) && outLength > 0 {
+		t.Errorf("error, input files length should match output length")
+	}
+
+	// Check the input metadata is the same as that output from the first module
+	inMetaData, err := ioutil.ReadFile(inMetaFilePath)
+	if err != nil {
+		t.Errorf("error reading in meta file '%s': '%+v'", inMetaFilePath, err)
+	}
+
+	var kvps common.KeyValuePairs
+	err = json.Unmarshal(inMetaData, &kvps)
+	if err != nil {
+		t.Errorf("error decoding file '%s' content: '%+v'", inMetaFilePath, err)
+	}
+
+	// The first key, value pair should be as expected
+	for _, kvp := range kvps {
+		if kvp.Key != "abc" {
+			t.Errorf("expected key 'abc' in key value pairs: '%+v'", kvp)
+		}
+		if kvp.Value != "123" {
+			t.Errorf("expected key 'abc' to have value '123' in key value pairs: '%+v'", kvp)
+		}
+		break
+	}
+
+	// Clean up external resources
+	_ = os.RemoveAll(inEventsDir)
```

Comment:
Is it worth doing the cleanup step as a `defer` statement earlier in the tests. That would ensure that if the test crashed mid-run when rerun it would have a clean environment?

Reply:
good spot - fixed in latest commit.